02_bubble/alextmz/main.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) []int {
+	r := make([]int, len(s))
+	copy(r, s)
+
+	for {
+		sorted := true
+		for i := 1; i < len(r); i++ {
```

Comment:
Still not bubble sort as I understand it, sorry ;)
My understanding of bubble sort is: 
you compare adjacent elements and if left bigger than right you swap. Doing so will have the largest element in the right most position of the slice (at position `len(r)-1`) when the first outer iteration is completed. You therefore only need to sort to `len(r)-2` in the next outer iteration and so forth.... The loop indices in bubble sort according to my understanding should run:
``` go
for i := len(s); i > 0; i-- {
		for j := 1; j < i; j++ {
```
If I read your code correctly you resort to the end on every outer iteration even though you already know that the rightmost `n` elements are sorted, with `n` being the count of the outer loop.